### PR TITLE
Horrible bug fix :P

### PR DESCRIPTION
**Pull request by digetx**
This fixes Xorg crash / hang on the start of x11perf tests.

### CODE REVIEW

**Comment by digetx** [comment]
Interestingly, under x11perf I'm getting quite many of
> (EE) tegra_stream.c:155/tegra_stream_submit(): drm_tegra_job_submit() failed -22
> (EE) tegra_stream.c:186/tegra_stream_put_fence(): drm_tegra_fence_wait_timeout() failed -22

Still something else is wrong, but maybe it's Opentegra's issue.

**Comment by kusma** [comment]
I've gotten those in Mesa also. But I haven't tried with a very up-to-date version...

**Comment by kusma** [comment]
s/valgring/valgrind/ btw ;)

File: tegra/pushbuf.c [hunk 1]
```diff
@@ -167,7 +167,7 @@ int drm_tegra_pushbuf_prepare(struct drm_tegra_pushbuf *pushbuf,
 	DRMLISTADD(&bo->push_list, &priv->bos);
 
 	priv->start = priv->base.ptr = ptr;
-	priv->end = priv->start + bo->size;
+	priv->end = priv->start + bo->size / sizeof(uint32_t);
```
Comment (kusma) [inline tegra/pushbuf.c]: Yikes!

Comment (digetx) [inline tegra/pushbuf.c]: Valgrind doesn't catch it for some reason.

**Comment by kusma** [comment]
This broke the build for me:
```
tegra.c: In function 'drm_tegra_bo_map':
tegra.c:292:6: error: 'RUNNING_ON_VALGRIND' undeclared (first use in this function)
  if (RUNNING_ON_VALGRIND) {
      ^~~~~~~~~~~~~~~~~~~
tegra.c:292:6: note: each undeclared identifier is reported only once for each function it appears in
tegra.c:298:4: error: implicit declaration of function 'VALGRIND_MAKE_MEM_NOACCESS' [-Werror=implicit-function-declaration]
    VALGRIND_MAKE_MEM_NOACCESS(bo->map, bo->size);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~
tegra.c:298:4: warning: nested extern declaration of 'VALGRIND_MAKE_MEM_NOACCESS' [-Wnested-externs]
tegra.c:305:4: error: implicit declaration of function 'VALGRIND_MAKE_MEM_DEFINED' [-Werror=implicit-function-declaration]
    VALGRIND_MAKE_MEM_DEFINED(bo->map, bo->size);
    ^~~~~~~~~~~~~~~~~~~~~~~~~
tegra.c:305:4: warning: nested extern declaration of 'VALGRIND_MAKE_MEM_DEFINED' [-Wnested-externs]
tegra.c: In function 'drm_tegra_bo_unmap':
tegra.c:333:6: error: 'RUNNING_ON_VALGRIND' undeclared (first use in this function)
  if (RUNNING_ON_VALGRIND && bo->mmap_ref == 2)
      ^~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [Makefile:671: tegra.lo] Error 1
make[1]: *** [Makefile:819: all-recursive] Error 1
make: *** [Makefile:578: all] Error 2
```

**Comment by kusma** [comment]
Fix here: #15

**Pull request merged**